packages/editor/src/bindings/post-meta.js

```diff
@@ -51,15 +51,20 @@ function getPostMetaFields( select, context ) {
 	const registeredFields = getRegisteredPostMeta( context?.postType );
 	const metaFields = {};
 	Object.entries( registeredFields || {} ).forEach( ( [ key, props ] ) => {
-		// Don't include footnotes or private fields.
-		if ( key !== 'footnotes' && key.charAt( 0 ) !== '_' ) {
+		if (
+			// Don't include footnotes.
+			key !== 'footnotes' &&
+			// Don't include private fields.
+			key.charAt( 0 ) !== '_'
+		) {
 			metaFields[ key ] = {
 				label: props.title || key,
 				value:
 					// When using the entity value, an empty string IS a valid value.
 					entityMetaValues?.[ key ] ??
 					// When using the default, an empty string IS NOT a valid value.
 					( props.default || undefined ),
+				type: props.type,
```

Comment:
Thanks a lot for the wrap-up 🙂 

I think it could make sense to work on unifying the typing of block attributes and register meta somehow. For example, if in the future a `URL` type is added, it should be added to both. It is outside of the scope of this PR, but something to consider.

Reply:
This nicely ties in with the feedback shared by @afercia at https://github.com/WordPress/gutenberg/issues/63018#issuecomment-2405241927. In this PR, we will see the first step toward filtering of the options presented for the block attribute based on its type that matches with the Post Meta's type. The next step would be offering more granular type system for all meta and block attributes. We will have to enforce defining the type for custom block binding sources in the future, too.